File: nature-harvest-frontend/src/app/dtos/order/order.dto.ts
import { CartDto } from '../cart/cart.dto';

export class OrderDto {
  userId: string;
  email: string;
  name: string;
  phone: string;
  deliveryAddress: string;
  note: string;
  paymentMethod: string;
  amount: number;
  cartItems: CartDto[];

  constructor(data: any) {
    this.userId = data.userId;
    this.email = data.email;
    this.name = data.name;
    this.phone = data.phone;
    this.deliveryAddress = data.deliveryAddress;
    this.note = data.note;
    this.paymentMethod = data.paymentMethod;
    this.amount = data.amount;
    this.cartItems = data.cartItems;
  }
}

File: nature-harvest-backend/nature-harvest-backend/src/main/java/com/api/nature_harvest_backend/dtos/OrderDto.java
package com.api.nature_harvest_backend.dtos;

import lombok.*;

import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class OrderDto {
    private String userId;
    private String email;
    private String name;
    private String phone;
    private String deliveryAddress;
    private String note;
    private String paymentMethod;
    private int amount;
    private List<CartDto> cartItems;

}
